### PR TITLE
[SPARK-33680][SQL][TESTS][FOLLOWUP] Fix more test suites to have explicit confs

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This is a follow-up for SPARK-33680 to remove the assumption on the default value of `spark.sql.adaptive.enabled` .

### Why are the changes needed?

According to the test result https://github.com/apache/spark/pull/30628#issuecomment-739866168, the [previous run](https://github.com/apache/spark/pull/30628#issuecomment-739641105) didn't run all tests.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -404,7 +404,7 @@ object SQLConf {
       "middle of query execution, based on accurate runtime statistics.")
     .version("1.6.0")
     .booleanConf
-    .createWithDefault(false)
+    .createWithDefault(true)
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:407]: This will be reverted before merging.

**Review by HyukjinKwon — COMMENTED** [review]
I thought some tests failed in SparkR and PySpark too when its enabled ... but  looks like I was remembering wrongly. I'm good with this change. Cc @maryannxue and @cloud-fan

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon . Yes, PySpark seems to be okay while SparkR has some issue. 
However, SparkR currently has some flaky Watermark failure. So, I'll proceed SparkR separately after finishing.

**Comment by dongjoon-hyun** [comment]
Merged to master/3.1.

**Pull request closed without merging**